From 64114e84d0324e50ac7ea62c884f11a112f75ace Mon Sep 17 00:00:00 2001
From: Madhurkiran Harikrishnan <madhurkiran.harikrishnan@xilinx.com>
Date: Tue, 3 Mar 2020 15:14:44 -0800
Subject: [PATCH 1258/1852] drm/fb_helper: Enable drm_leak_fbdev_smem by
 default for zynqmp

commit 8ed205bdf58e3830d6e94612b95d9c18c679c458 from
https://github.com/Xilinx/linux-xlnx.git

As Mali400 user-space implementation relies on the physical address to
render into framebufffer and lot of use-case depends on this backend,
enable the module parameter 'drm_leak_fbdev_smem' by default.

The feature is deprecated in upstream due to the security concern, but
no alternative solution is given. For more information, refer to kernel
commit 4be9bd10e22d ("drm/fb_helper: Allow leaking fbdev smem_start").

This patch is Xilinx specific and not upstreamable.

Signed-off-by: Madhurkiran Harikrishnan <madhurkiran.harikrishnan@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/drm_fb_helper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/drm_fb_helper.c b/drivers/gpu/drm/drm_fb_helper.c
index 1124be028ff5..c3b5ed8eec4d 100644
--- a/drivers/gpu/drm/drm_fb_helper.c
+++ b/drivers/gpu/drm/drm_fb_helper.c
@@ -69,7 +69,7 @@ MODULE_PARM_DESC(drm_fbdev_overalloc,
  * considered as a broken and legacy behaviour from a modern fbdev device.
  */
 #if IS_ENABLED(CONFIG_DRM_FBDEV_LEAK_PHYS_SMEM)
-static bool drm_leak_fbdev_smem = false;
+static bool drm_leak_fbdev_smem = true;
 module_param_unsafe(drm_leak_fbdev_smem, bool, 0600);
 MODULE_PARM_DESC(drm_leak_fbdev_smem,
 		 "Allow unsafe leaking fbdev physical smem address [default=false]");
-- 
2.31.1

